src/Nervos/Address.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Bech32.h"
+#include "../Coin.h"
+
+#include <TrustWalletCore/TWCoinType.h>
+
+using namespace TW::Nervos;
+using namespace TW;
+
+bool Address::isValid(const std::string& string) {
+    return Address::isValid(string, TW::p2pkhPrefix(TWCoinTypeNervos),
+                            TW::p2shPrefix(TWCoinTypeNervos),
+                            stringForHRP(TW::hrp(TWCoinTypeNervos)));
+}
+
+bool Address::isValid(const std::string& string, byte p2pkh, byte p2sh, const char* hrp) {
+    try {
+        Address(string, p2pkh, p2sh, hrp);
+        return true;
+    } catch (std::exception e) {
+        return false;
+    }
+}
+
+Address::Address(const std::string& string)
+    : Address(string, TW::p2pkhPrefix(TWCoinTypeNervos), TW::p2shPrefix(TWCoinTypeNervos),
+              stringForHRP(TW::hrp(TWCoinTypeNervos))) {}
+
+Address::Address(const std::string& string, byte p2pkh, byte p2sh, const char* hrp) : hrp(hrp) {
+    bool isValid = false;
+
+    do {
```

Comment:
`do-while` is not justified here consider using a functor that throws in case of an error

Reply:
Done